examples/other/image_histogram.py

```diff
@@ -0,0 +1,263 @@
+)
+histogram_object = gfx.Image(
+    gfx.Geometry(grid=histogram_texture),
+    gfx.ImageBasicMaterial(clim=(0, 255), interpolation="nearest"),
+)
+histogram_object.local.y += 10
+histogram_object.local.scale_x = 512 / nbins
+scene.add(histogram_object)
+
+
+# Update camera to show the image and histogram
+camera.show_object(scene)
+
+
+## GPU Compute
+
+histogram_wgsl = """
+
+@group(0) @binding(0) var imageTexture: texture_2d<f32>;
+@group(0) @binding(1) var histTexture: texture_storage_2d<rgba8unorm,write>;
+
+// from: https://www.w3.org/WAI/GL/wiki/Relative_luminance
+const kSRGBLuminanceFactors = vec3f(0.2126, 0.7152, 0.0722);
+fn srgbLuminance(color: vec3f) -> f32 {
+    return saturate(dot(color, kSRGBLuminanceFactors));
+}
+
+@compute @workgroup_size(1)
+fn main() {
+    let size = textureDimensions(imageTexture, 0);
+    //let hist_size = textureDimensions(histTexture, 0);
+    let numBins = 20;//hist_size.x;
+    let numLevels = 100;//hist_size.y;
+
+    // Compute hist
+
+    var bins: array<i32,20>;
+
+    let lastBinIndex = u32(numBins - 1);
+    for (var y = 0u; y < size.y; y++) {
+        for (var x = 0u; x < size.x; x++) {
+        let position = vec2u(x, y);
+        let color = textureLoad(imageTexture, position, 0);
+        let v = srgbLuminance(color.rgb);
+        let bin = min(u32(v * f32(numBins)), lastBinIndex);
+        bins[bin] += 1;
+        }
+    }
+
+    // Draw hist
+
+    // Detect maximum count
+    var maxCount = 0;
+    for (var x = 0; x < numBins; x++) {
+        maxCount = max(maxCount, bins[x]);
+    }
+
+    // Fill output image
+    for (var x = 0; x < numBins; x++) {
+        let count = bins[x];
+        let max_y = i32(round( f32(numLevels) * f32(count) / f32(maxCount) ));
+        for (var y = 0; y < max_y; y++) {
+            let value = vec4<f32>(1.0, 1.0, 0.0, 1.0);
+            textureStore(histTexture, vec2<i32>(x, y), value);
+        }
+        for (var y = max_y; y < numLevels; y++) {
+            let value = vec4<f32>(0.0, 0.0, 0.0, 0.0);
+            textureStore(histTexture, vec2<i32>(x, y), value);
+        }
+
+    }
+}
+
+"""
+
+
+def compute_histogram():
+    nx, ny, nz = 100, 1, 1
+
+    # Get Pygfx's wgu device object
+    device = gfx.renderers.wgpu.Shared.get_instance().device
+
+    # TODO: accessing internal _wgpu_object attribute
+    image_wgpu_texture = image_object.geometry.grid._wgpu_object
+    histogram_wgpu_texture = histogram_texture._wgpu_object
+
+    # Compile our shader
+    # TODO: don't do this every time!
+    cshader = device.create_shader_module(code=histogram_wgsl)
+    compute = {
+        "module": cshader,
+        "entry_point": "main",
+    }
+    # compute["constants"] = constants
+
+    # Create a pipeline
+    compute_pipeline = device.create_compute_pipeline(
+        layout="auto",
+        compute=compute,
+    )
+
+    # Create bindings
+    bindings = [
+        {
+            "binding": 0,
+            "resource": image_wgpu_texture.create_view(
```

Comment:
Wait so when this is done, is it sharing the same buffer for rendering the Texture as an image shown on the screen as well as sharing this buffer for the compute shader? It's that's easy? I have so many ideas!

Reply:
Turns out compute shaders are very useful! 😁

Reply:
But also the other way around. Let ppl generate a texture using GPU compute, and the just show it in Pygfx. Same for buffers that can be used for positons, colors, etc.

Reply:
> Let ppl generate a texture using GPU compute, and the just show it in Pygfx

Yes definitely something I'm interested in!